qiling/loader/elf.py

```diff
@@ -87,6 +88,9 @@ def run(self):
         stack_size = self.profile.getint('stack_size')
         self.ql.mem.map(stack_address, stack_size, info='[stack]')
 
+        # Setup heap
+        self.ql.os.heap = QlMemoryHeap(self.ql, 0x3000000, 0x3000000 + 0x1000000)
```

Comment:
Heap base and size should be set based on the value configured in the profile file. Using hardcoded values is not a good idea.